src/main/asciidoc/index.adoc

```diff
@@ -0,0 +1,426 @@
+
+For example, the next snippet is a valid Groovy verticle:
+
+[source, groovy]
+----
+println "Hello from vertx"
+----
+
+On deployment, by default, Vert.x executes the script. Optionally, your script can provide the `startVertx` and
+`stopVertx` methods. Theses methods are called respectively when the verticle starts and stops:
+
+[source, groovy]
+----
+void vertxStart() {
+ println "starting"
+}
+
+void vertxStop() {
+ println "stopping"
+}
+----
+
+Alternatively, you can extend the {@link io.vertx.core.AbstractVerticle} class and implement the `start` and
+`stop` methods:
+
+[source, groovy]
+----
+import io.vertx.core.AbstractVerticle;
+
+class HelloWorldHttpVerticle extends AbstractVerticle {
+
+ void start() {
+  println("Starting")
+ }
+
+ void stop() {
+   println("Stopping")
+ }
+}
+----
+
+When Vert.x deploys the verticle it will call the `start` method, and when the method has completed the
+verticle will be considered started.
+
+You can also optionally override the `stop` method. This will be called by Vert.x when the verticle is undeployed
+and when the method has completed the verticle will be considered stopped.
+
+=== Accessing the vertx instance from a verticle
+
+Regardless the way you use to implement your verticle, you access the vert.x instance using the `vertx` variable /
+field.
+
+.Access to the vert.x instance in a Groovy script
+[source, groovy]
+----
+vertx.deployVerticle("another_verticle.groovy")
+----
+
+.Access to the vert.x instance in a Groovy class
+[source, groovy]
+----
+import io.vertx.lang.groovy.GroovyVerticle;
+
+class HelloWorldHttpVerticle extends GroovyVerticle {
+
+   void start() {
+     vertx.deployVerticle("another_verticle.groovy")
+   }
+}
+----
+
+=== Asynchronous Verticle start and stop
+
+Sometimes you want to do something in your verticle start-up which takes some time and you don't want the verticle to
+be considered deployed until that happens. For example you might want to deploy other verticles in the start method.
+
+You can't block waiting for the other verticles to deploy in your start method as that would break the Golden Rule.
+
+So how can you do this?
+
+The way to do it is to implement the *asynchronous* start method. This version of the method takes
+a {@link io.vertx.core.Promise} as a parameter. When the method returns the verticle will *not* be considered deployed.
+
+Some time later, after you've done everything you need to do (e.g. start other verticles), you can call complete
+on the Promise (or fail) to signal that you're done. Similarly, there is an asynchronous version of the stop method too.
+
+You use this if you want to do some verticle cleanup that takes some time.
+
+When your Verticle is implemented as a script, asynchronous start and stop are implemented as follows:
+
+[source, groovy]
+----
+import io.vertx.core.Future
+
+void vertxStart(Future<Void> future) {
```

Comment:
Should be Promise here, right?

Reply:
yes I fixed it, and I changed also the implementation that was still considering `Future`. Normally it should not have been working but given that Groovy is dynamic, it just worked. (FutureImpl implements Future and Promise)